src/exporter/DaemonMetricCollector.cc

```diff
@@ -291,18 +291,34 @@ DaemonMetricCollector::get_labels_and_metric_name(std::string daemon_name,
   std::string new_metric_name;
   labels_t labels;
   new_metric_name = metric_name;
+  const std::string ceph_daemon_prefix = "ceph-";
+  const std::string ceph_client_prefix = "client.";
+  if (daemon_name.rfind(ceph_daemon_prefix, 0) == 0) {
+    daemon_name = daemon_name.substr(ceph_daemon_prefix.size());
+  }
+  if (daemon_name.rfind(ceph_client_prefix, 0) == 0) {
+    daemon_name = daemon_name.substr(ceph_client_prefix.size());
+  }
   // In vstart cluster socket files for rgw are stored as radosgw.<instance_id>.asok
   if (daemon_name.find("radosgw") != std::string::npos) {
     std::size_t pos = daemon_name.find_last_of('.');
     std::string tmp = daemon_name.substr(pos+1);
     labels["instance_id"] = quote(tmp);
   }
   else if (daemon_name.find("rgw") != std::string::npos) {
-    std::string tmp = daemon_name.substr(16, std::string::npos);
-    std::string::size_type pos = tmp.find('.');
-    labels["instance_id"] = quote("rgw." + tmp.substr(0, pos));
+    // fetch intance_id for e.g. "okbvtv" from daemon_name=rgw.foo.ceph-node-00.okbvtv 
+    std::vector<std::string> elems;
+    std::stringstream ss;
+    ss.str(daemon_name);
+    std::string item;
+    while (std::getline(ss, item, '.')) {
+        elems.push_back(item);
+    }
+    labels["instance_id"] = quote(elems[3]);
```

Comment:
I don't think a vector of strings is needed to only retrieve the last one of the values on a '.' delimitered list. You can just count the last value and use it. Maybe there could be a out of range error in elems[3].

Reply:
Ah yes! Thanks for noticing! it makes things much simpler, will update it.

Reply:
Done!

Reply:
We actually need to fetch the 4th element only as this fails if daemon name is like `ceph-client.rgw.foo.ceph-node-00.hrgsea.2.94739968030880`

Reply:
I've opened a PR to bring back these changes.